test/Microsoft.ML.Tests/Scenarios/ClusteringTests.cs

```diff
@@ -0,0 +1,121 @@
+
+Until the day your dog can talk, you'll never likely hear him pronounce ""I love you,"" and in the meantime, don't expect him to purchase you a Hallmark card or some balloons with those renowned romantic words printed on top. Also, don’t expect a box of chocolates or a bouquet of flowers from your dog when Valentine's day is around the corner. Sometimes it might feel like we're living an uneven relationship, but just because dogs don't communicate their love the way we do, doesn't mean they don't love us!"
+            });
+        }
+
+        public class NewsData
+        {
+            [Column(ordinal: "0")]
+            public string Id;
+
+            [Column(ordinal: "1", name: "Label")]
+            public string Topic;
+
+            [Column(ordinal: "2")]
+            public string Subject;
+
+            [Column(ordinal: "3")]
+            public string Content;
+        }
+
+        public class ClusteringPrediction
+        {
+            [ColumnName("PredictedLabel")]
+            public uint SelectedClusterId;
+            [ColumnName("Score")]
+            public float[] Distance;
+        }
+
+        public class ClusteringData
+        {
+            [ColumnName("Features")]
+            [VectorType(2)]
+            public float[] Points;
+        }
+
+        [Fact]
+        public void PredictClusters()
+        {
+            int n = 1000;
+            int k = 5;
+            var rand = new Random();
+            var clusters = new ClusteringData[k];
+            var data = new ClusteringData[n];
+            for (int i = 0; i < k; i++)
+            {
+                //pick clusters as points on circle with angle to axis X equal to 360*i/k
+                clusters[i] = new ClusteringData { Points = new float[2] { (float)Math.Cos(Math.PI * i * 2 / k), (float)Math.Sin(Math.PI * i * 2 / k) } };
+            }
+            // create data points by randomly picking cluster and shifting point slightly away from it.
+            for (int i = 0; i < n; i++)
+            {
+                var index = rand.Next(0, k);
+                var shift = (rand.NextDouble() - 0.5) / k;
+                data[i] = new ClusteringData
+                {
+                    Points = new float[2]
+                    {
+                        (float)(clusters[index].Points[0] + shift),
+                        (float)(clusters[index].Points[1] + shift)
+                    }
+                };
+            }
+            var pipeline = new LearningPipeline();
+            pipeline.Add(CollectionDataSource.Create(data));
+            pipeline.Add(new KMeansPlusPlusClusterer() { K = k });
+            var model = pipeline.Train<ClusteringData, ClusteringPrediction>();
+            //validate no initial centers of clusters belong to same cluster.
```

Comment:
Just a minor comment.
`//validate no initial centers of clusters belong to same cluster.`
These don't seem to be `initial center` as these are not set as initial cluster centers to KMean trainer. That is what initial center means in KMean or other clustering algorithms. 

Rather, these are just a data points curated in a way that these appear to be cluster centers initially.

Reply:
is current phrasing better?

---
In reply to: [190697440](https://github.com/dotnet/machinelearning/pull/222#discussion_r190697440) [](ancestors = 190697440)